tensorflow/java/src/main/java/org/tensorflow/Tensor.java

```diff
@@ -34,6 +41,115 @@
  * }</pre>
  */
 public final class Tensor implements AutoCloseable {
+
+  /**
+   * Create a Tensor with data from the given buffer.
+   *
+   * <p>Provide a {@link ByteBuffer} to create a Tensor of any datatype.  Note that primitive
+   * data must be in native byte order. Provide a {@link FloatBuffer}, {@link IntBuffer}, {@link DoubleBuffer},
+   * or {@link LongBuffer} to create a Tensor with a datatype of {@code FLOAT}, {@code INT32}, {@code DOUBLE},
+   * or {@code INT64} respectively.
+   *
+   * <p> This method copies
+   * <i>n</i>&nbsp;=&nbsp;<tt>src.remaining()</tt> elements from the given
+   * buffer into the tensor, starting at the buffer's current position.
+   * The position of the buffer is then incremented by <i>n</i>.
+   *
+   * @param dataType the tensor datatype.
+   * @param shape the tensor shape.
+   * @param data a buffer containing the tensor data.
+   */
+  public static Tensor create(DataType dataType, long[] shape, Buffer data) {
```

Comment:
Believe it or not, my first draft was entirely `ByteBuffer` based.   I found it necessary to support other `Buffer` types to optimize the transfer of wrapped primitive arrays. 

Consider a `FloatBuffer` as produced by `FloatBuffer.wrap(new float[] { ... })`.  It is not convertible to a `ByteBuffer`.   To write it directly to the tensor's native `ByteBuffer`, we call `put` on a view of the latter.   At all times we avoid exposing the native buffer to user code.

I'm fine with moving to a set of overloaded methods to retain the benefits of other buffer types while reducing the possibility of runtime exceptions.